io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -578,7 +597,31 @@ public Instance next() {
 			}
 
 			Instance instance = iterator.next();
-			return new StreamGmlInstance(instance, totalFeaturesProcessed++);
+
+			Iterable<QName> mapPropertiess = instance.getPropertyNames();
+			for (QName qName : mapPropertiess) {
+				if (qName.getLocalPart().equals("id") && qName.getPrefix().equals("gml")) {
```

Comment:
The namespace prefix is nothing that should be checked, because that can be an arbitrary value.
[Here](https://github.com/halestudio/hale/blob/2baec50f5ba9117f09942c13725b031a449c17f0/io/plugins/eu.esdihumboldt.hale.io.wfs/src/eu/esdihumboldt/hale/io/wfs/file/WFSFeatureCollectionWriter.java#L93) is an example where we check for a GML namespace.

Since the GML namespace used should be the same in the same data, I would suggest to only determine the property name once and after that use a more efficient approach than iterating all property names by getting the property directly.

Reply:
There is something I don't understand. 
In the GmlInstanceCollection the [TypeDefinition def](https://github.com/halestudio/hale/blob/2baec50f5ba9117f09942c13725b031a449c17f0/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/GmlInstanceCollection.java#L170) actually have a namespace URI starting with http://www.opengis.net/wfs,
but when arriving to the [WfsBakendGmlInstanceCollection](https://github.com/halestudio/hale/blob/2baec50f5ba9117f09942c13725b031a449c17f0/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java#L573) the instance doesn't contain anymore a namespace URI starting with http://www.opengis.net/wfs, but just a QName; id, http://www.opengis.net/gml, gml, so cannot even be compared with a namespace such as http://www.opengis.net/wfs.